Libraries/pbxspec/Sources/PBX/ProductType.cpp

```diff
@@ -195,6 +219,61 @@ parse(Context *context, plist::Dictionary const *dict, std::unordered_set<std::s
         _requiredBuiltProductsDir = RBPD->value();
     }
 
+    if (ABPs != nullptr) {
+        _allowedBuildPhases = std::vector<std::string>();
+        for (size_t n = 0; n < ABPs->count(); n++) {
+            if (auto ABP = ABPs->value<plist::String>(n)) {
+                _allowedBuildPhases->push_back(ABP->value());
+            }
+        }
+    }
+
+    if (AFTs != nullptr) {
+        _allowedFileTypes = std::unordered_map<std::string, std::vector<std::string>>();
+        for (size_t n = 0; n < AFTs->count(); n++) {
+            if (auto AFT = AFTs->value<plist::Array>(n)) {
+                auto result = _allowedFileTypes->insert({ AFTs->key(n), std::vector<std::string>() });
+
+                for (size_t m = 0; m < AFT->count(); m++) {
+                    if (auto FT = AFT->value<plist::String>(m)) {
+                        std::vector<std::string> *allowedFileTypes = &result.first->second;
+                        allowedFileTypes->push_back(FT->value());
+                    }
+                }
+            }
+        }
+    }
+
+    if (BPFRAs != nullptr) {
+        _buildPhaseFileRefAdditions = std::unordered_map<std::string, std::vector<FileReference>>();
+        for (size_t n = 0; n < BPFRAs->count(); n++) {
+            if (auto BPFRA = BPFRAs->value<plist::Array>(n)) {
```

Comment:
Yeah, in theory. In practice, don't really have error reporting setup in a way that won't get annoying.